src/mlpack/bindings/R/CMakeLists.txt

```diff
@@ -0,0 +1,217 @@
+macro (not_found_return message)
+  message(STATUS "${message}")
+  macro (add_r_binding name)
+    # Do nothing.
+  endmacro ()
+
+  return()
+endmacro ()
+
+# If we are not supposed to make R bindings, define the macro so it does
+# nothing and leave this file.
+if (NOT BUILD_R_BINDINGS)
+  not_found_return("Not building R bindings.")
+endif ()
+
+if (BUILD_R_BINDINGS)
+  ## We need to check here if R is even available.  Although actually
+  ## technically, I'm not sure if we even need to know!  For the tests though we
+  ## do.  So it's probably a good idea to check.
+  if (FORCE_BUILD_R_BINDINGS)
+    find_package(R 3.2)
+    if (NOT R_FOUND)
+      unset(BUILD_R_BINDINGS CACHE)
+      message(FATAL_ERROR "Could not Build R Bindings")
+    endif()
+  else ()
+    find_package(R 3.2)
+    if (NOT R_FOUND)
+      unset(BUILD_R_BINDINGS CACHE)
+    endif()
+  endif ()
+
+  if (NOT R_FOUND)
+    not_found_return("R not found; not building R bindings.")
+  endif ()
+
+  add_custom_target(R ALL)
+
+  # Now configure DESCRIPTION.
+  file(READ "${CMAKE_SOURCE_DIR}/src/mlpack/core/util/version.hpp"
+      VERSION_HPP_CONTENTS)
+  string(REGEX REPLACE ".*#define MLPACK_VERSION_MAJOR ([0-9]+).*" "\\1"
+      MLPACK_VERSION_MAJOR "${VERSION_HPP_CONTENTS}")
+  string(REGEX REPLACE ".*#define MLPACK_VERSION_MINOR ([0-9]+).*" "\\1"
+      MLPACK_VERSION_MINOR "${VERSION_HPP_CONTENTS}")
+  string(REGEX REPLACE ".*#define MLPACK_VERSION_PATCH [\"]?([0-9x]+)[\"]?.*"
+      "\\1" MLPACK_VERSION_PATCH "${VERSION_HPP_CONTENTS}")
+  set(PACKAGE_VERSION
+      "${MLPACK_VERSION_MAJOR}.${MLPACK_VERSION_MINOR}.${MLPACK_VERSION_PATCH}")
+   
+  execute_process(
+    COMMAND ${PYTHON_EXECUTABLE} -c 
+      "from time import gmtime, strftime; print(strftime(\"%Y-%m-%d\", gmtime()))"
+      OUTPUT_VARIABLE PACKAGE_DATE
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+   )
```

Comment:
I didn't notice this until now, but we're not guaranteed that Python will be available on the system we're using.  I think we can just use `string(TIMESTAMP ...)`: https://stackoverflow.com/questions/24292898/compile-date-and-time-using-cmake

Reply:
Updated.